src/query/plan/operator.cpp

```diff
@@ -1043,17 +1059,23 @@ class ExpandVariableCursor : public Cursor {
       // if we are here, we have a valid stack,
       // get the edge, increase the relevant iterator
       auto current_edge = *edges_it_.back()++;
+      if (context.auth_checker && !context.auth_checker->Accept(context.user, *context.db_accessor, current_edge.first))
+        continue;
 
       // Check edge-uniqueness.
       bool found_existing =
           std::any_of(edges_on_frame.begin(), edges_on_frame.end(),
                       [&current_edge](const TypedValue &edge) { return current_edge.first == edge.ValueEdge(); });
       if (found_existing) continue;
-
-      AppendEdge(current_edge.first, &edges_on_frame);
       VertexAccessor current_vertex =
           current_edge.second == EdgeAtom::Direction::IN ? current_edge.first.From() : current_edge.first.To();
 
+      if (context.auth_checker &&
+          !context.auth_checker->Accept(context.user, *context.db_accessor, current_vertex, storage::View::OLD))
+        continue;
+
+      AppendEdge(current_edge.first, &edges_on_frame);
+
```

Comment:
why is this moved, I know that Benjo commented something on this one, but what was the initial ordering? (I'm not sure where you first introduced this, in edge type filtering or here?)

Reply:
Ordering before was AppendEdge then get current_vertex but here I don't want to AppendEdge on frame if vertex I'm expanding to is not authorized.